src/app/common/hooks/use-background-location-redirect.ts

```diff
@@ -0,0 +1,34 @@
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+import { RouteUrls } from '@shared/route-urls';
+
+import { useLocationState } from '@app/common/hooks/use-location-state';
```

Comment:
I still need to refactor this as per: https://github.com/hirosystems/wallet/pull/4062#discussion_r1279091163